provider/image.go

```diff
@@ -252,11 +253,18 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 		}
 	}
 
+	dist, err := docker.DistributionInspect(ctx, img.Name, authConfigEncoded)
+	if err != nil {
+		return "", nil, err
+	}
+	repoDigest := fmt.Sprintf("%s@%s", img.Name, dist.Descriptor.Digest)
```

Comment:
What would you like to see happen if we fail here because the registry is inaccessible, we have push but not get permissions, etc.?

I don't think it's a good option for us to fail the create/update if the push succeeds but we cannot obtain a digest.

Reply:
> we have push but not get permissions

Is that a realistic scenario? At least with Docker Hub I don't think it's possible to have write but not read!

> I don't think it's a good option for us to fail the create/update if the push succeeds but we cannot obtain a digest.

At least at @MaterializeInc we'd prefer to loudly fail the create/update with the error fetching the digest (as done here), rather than trying to limp along with a `nil` repo digest. Given that historical versions of this resource included the digest, it seems like empirically this approach didn't cause trouble for folks.

That said, if there are other users who really can't tolerate that, I think it works okay for our use case to return `nil` for the repo digest, as long as there is a clear diagnostic emitted with the digest fetch error. If that's the desired path, I could use help writing that code, as I'm no longer familiar enough with Pulumi providers to remember how to emit that sort of diagnostic.

Reply:
Could we try to use ImageInspectWithRaw instead of DistributionInspect?

Reply:
Sure—are you comfortable with this bit of logic to find the repo digest that matches the image name?

https://github.com/MaterializeInc/pulumi-docker-buildkit/blob/ea3b497dfd3bd76f4525a79ec90b4c7a4d8aa80b/cmd/pulumi-resource-docker-buildkit/main.go#L315-L325

Reply:
This client RPC appears to be exactly what `docker image inspect` performs.